api/src/main/java/org/infinispan/api/common/CacheEntryExpiration.java

```diff
@@ -0,0 +1,47 @@
+package org.infinispan.api.common;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * @since 14.0
+ **/
+public class CacheEntryExpiration {
+   public static final CacheEntryExpiration IMMORTAL = new CacheEntryExpiration();
+   private final Duration lifespan;
+   private final Duration maxIdle;
+
+   private CacheEntryExpiration() {
+      this.lifespan = null;
+      this.maxIdle = null;
+   }
+
+   private CacheEntryExpiration(Duration lifespan, Duration maxIdle) {
+      this.lifespan = lifespan;
+      this.maxIdle = maxIdle;
+   }
+
+   public static CacheEntryExpiration withLifespan(Duration lifespan) {
+      return new CacheEntryExpiration(lifespan, null);
+   }
+
+   public static CacheEntryExpiration withMaxIdle(Duration maxIdle) {
+      return new CacheEntryExpiration(null, maxIdle);
+   }
+
+   public static CacheEntryExpiration withLifespanAndMaxIdle(Duration lifespan, Duration maxIdle) {
+      return new CacheEntryExpiration(lifespan, maxIdle);
+   }
+
+   public Optional<Duration> lifespan() {
+      return Optional.ofNullable(lifespan);
+   }
+
+   public Optional<Duration> maxIdle() {
+      return Optional.ofNullable(maxIdle);
+   }
+
+   public boolean isImmortal() {
+      return lifespan == null && maxIdle == null;
```

Comment:
I would change the static factories to check for `null` and return the `IMMORTAL` instance.
This code could be changed to `this == IMMORTAL`

Reply:
Done